File: 0069-sqrtx/Solution.go
package Leetcode

func mySqrt(x int) int {
	left := 0
	right := x
	for left <= right {
		mid := left + (right-left)/2
		if mid*mid > x {
			right = mid - 1
		} else if mid*mid < x {
			left = mid + 1
		} else {
			return int(mid)
		}
	}
	return int(right)

}

File: 0069-sqrtx/Solution.cs
public class Solution {
    public int MySqrt(int x)
    {
        Int64 left = 0;
        Int64 right = x;
        while (left <= right)
        {
            Int64 mid = left + (right - left) / 2;
            if (mid * mid == x)
            {
                return (int)mid;
            }
            else if (mid * mid < x)
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
        }
        return (int)right;
    }
}